tests/unit/providers/adm/conftest.py

```
@@ -0,0 +1,66 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Module for test configurations for the AdM provider unit test directory."""
+from typing import Any
+
+import pytest
+from pytest_mock import MockerFixture
+
+from merino.providers.adm.backends.protocol import AdmBackend, Content
+from merino.providers.adm.provider import Provider
+
+
+@pytest.fixture(name="adm_suggestion_content")
+def fixture_adm_suggestion_content() -> Content:
+    """Define backend suggestion content for test."""
+    return Content(
+        suggestions={
+            "firefox": (0, 0),
+            "firefox account": (0, 0),
+            "firefox accounts": (0, 0),
+            "mozilla": (0, 1),
+            "mozilla firefox": (0, 1),
+            "mozilla firefox account": (0, 1),
+            "mozilla firefox accounts": (0, 1),
+        },
+        full_keywords=["firefox accounts", "mozilla firefox accounts"],
+        results=[
+            {
+                "id": 2,
+                "url": "https://example.org/target/mozfirefoxaccounts",
+                "click_url": "https://example.org/click/mozilla",
+                "iab_category": "5 - Education",
+                "icon": "01",
+                "advertiser": "Example.org",
+                "title": "Mozilla Firefox Accounts",
+            }
+        ],
+        icons={1: "attachment-host/main-workspace/quicksuggest/icon-01"},
+    )
+
+
+@pytest.fixture(name="adm_parameters")
+def fixture_adm_parameters() -> dict[str, Any]:
+    """Define provider parameters for test."""
+    return {
+        "score": 0.3,
+        "score_wikipedia": 0.2,
+        "name": "adm",
+        "resync_interval_sec": 10800,
+    }
+
+
+@pytest.fixture(name="backend_mock")
+def fixture_backend_mock(mocker: MockerFixture, adm_suggestion_content: Content) -> Any:
```

Is there a need for using a mock here rather than defining custom backends that implement the protocol and then raise an exception (instead of `backend_mock.fetch.side_effect = Exception(error_message)`) or return a canned response (rather than `backend_mock.fetch.return_value = adm_suggestion_content`)?

We do that for Wikipedia in `merino/providers/wikipedia/backends/test_backends.py`.

---

I prefer mocking over custom backends, or fakes, here primarily because there is less boilerplate code to create and maintain. It also allows the response to be re-used for verification in this case.